From 2d315bbce860482b5ddb01a88fe9d0f01925998f Mon Sep 17 00:00:00 2001
From: Mario Schuknecht <mario.schuknecht@dresearch-fe.de>
Date: Mon, 18 May 2015 20:52:07 +0200
Subject: [PATCH] mxc_ipuv3_fb: Fix default bpp

So far bpp (bit per pixel) is 16 if not specified as kernel command line option.
This commit use bpp from device tree entry 'default_bpp' if no command line
option given.

Signed-off-by: Mario Schuknecht <mario.schuknecht@dresearch-fe.de>
---
 drivers/video/mxc/mxc_ipuv3_fb.c | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/drivers/video/mxc/mxc_ipuv3_fb.c b/drivers/video/mxc/mxc_ipuv3_fb.c
index 344f243..d331b40 100644
--- a/drivers/video/mxc/mxc_ipuv3_fb.c
+++ b/drivers/video/mxc/mxc_ipuv3_fb.c
@@ -2198,8 +2198,11 @@ static int mxcfb_dispdrv_init(struct platform_device *pdev,
 		mxcfbi->default_bpp = setting.default_bpp;
 
 		/* if "fbpix" was not set from boot command, use default value */
-		if(mxcfbi->fb_pix_fmt == 0)
+		if(mxcfbi->fb_pix_fmt == 0) {
 			mxcfbi->fb_pix_fmt = bpp_to_pixfmt(setting.default_bpp);
+			if (fb_pix_fmt)
+				pixfmt_to_var(fb_pix_fmt, &fbi->var);
+		}
 
 		ret = mxcfb_get_crtc(&pdev->dev, mxcfbi, setting.crtc);
 		if (ret)
@@ -2340,8 +2343,11 @@ static int mxcfb_option_setup(struct platform_device *pdev, struct fb_info *fbi)
 	if (fb_mode_str)
 		pdata->mode_str = fb_mode_str;
 
-	if(fb_pix_fmt == 0)
+	if(fb_pix_fmt == 0) {
 		fb_pix_fmt = bpp_to_pixfmt(pdata->default_bpp);
+		if (fb_pix_fmt)
+			pixfmt_to_var(fb_pix_fmt, &fbi->var);
+	}
 	pdata->fb_pix_fmt = fb_pix_fmt;
 
 	return 0;
-- 
2.1.4

